packages/context/src/context.ts

```diff
@@ -47,22 +47,32 @@ export class Context {
   bind<ValueType = BoundValue>(
     key: BindingAddress<ValueType>,
   ): Binding<ValueType> {
+    const binding = new Binding<ValueType>(key.toString());
+    this.add(binding);
+    return binding;
+  }
+
+  /**
+   * Add a binding to the context. If a locked binding already exists with the
+   * same key, an error will be thrown.
+   * @param binding The configured binding to be added
+   */
+  add<ValueType = BoundValue>(binding: Binding<ValueType>): this {
+    const key = binding.key;
     /* istanbul ignore if */
     if (debug.enabled) {
       debug('Adding binding: %s', key);
     }
-    key = BindingKey.validate(key);
```

Comment:
IIUC, this validation can be removed because `Binding`'s constructor has already validated the key 👍